docs/articles/advanced_topics/buttons.md

```diff
@@ -0,0 +1,117 @@
+>
+> Link buttons do **not** have a custom id and do **not** send interactions when pressed.
+
+Buttons consist of five parts:
+- Id
+- Style
+- Label
+- Emoji
+- Disabled
+
+The id of the button is a settable string on buttons, and is specified by the developer. Discord sends this id back in the [interaction object](https://discord.dev/interactions/slash-commands#interaction).
+
+The style of a button is one of four colors: Blurple, Grey, Green, and Red. These are referred to as Primary, Secondary, Success, and Danger respectively.
+
+The label of a button is optional *if* an emoji is specified. The label can be up to 80 characters in length. 
+The emoji of a button is a [partial emoji object](https://discord.dev/interactions/message-components#component-object), which means that **any valid emoji is usable**, even if your bot does not have access to it's origin server.
+
+The disabled field of a button is rather self explanatory. If this is set to true, the user will see a greyed out button which they cannot interact with. 
+
+# Making buttons
+
+Making buttons is relatively simple. Simply make a builder, and sprinkle some content and buttons on.
+
+
+```cs
+var builder = new DiscordMessageBuilder();
+
+builder
+    .WithContent("This message has buttons! Pretty neat innit?")
+    .WithComponents(new DiscordComponent[] 
+    {
+        new DiscordButtonComponent(ButtonStyle.Primary, "1_top" "Blurple!"),
+        new DiscordButtonComponent(ButtonStyle.Secondary, "2_top", "Grey!"),
+        new DiscordButtonComponent(ButtonStyle.Success, "3_top", "Green!"),
+        new DiscordButtonComponent(ButtonStyle.Danger, "4_top", "Red!"),
+        new DiscordButtonComponent("https://some-super-cool.site", "Link!")
+    })
+    .WithComponents(new DiscordComponent[] 
+    {
+        new DiscordButtonComponent(ButtonStyle.Primary, "1_top_d" "Blurple!", true),
+        new DiscordButtonComponent(ButtonStyle.Secondary, "2_top_d" "Grey!", true),
+        new DiscordButtonComponent(ButtonStyle.Success, "3_top_d", "Green!", true),
+        new DiscordButtonComponent(ButtonStyle.Danger, "4_top_d" "Red!", true),
+        new DiscordButtonComponent("https://some-super-cool.site", "Link!", true)
+    });
+
+await builder.SendAsync(someChannel);
+```
+As a note, custom ids can contain spaces. For these examples, underscores will be used. 
+
+Produces a message like such: ![Buttons](/images/advanced_topics_buttons_01.png)
+
+
+# Responding to button presses
+
+When any button is pressed, it will fire the [ComponentInteractionCreated](xref:DSharpPlus.DiscordClient#ComponentInteractionCreated) event.
+
+In the event args, `Id` will be the id of the button you specified. There's also an `Interaction` property, which contains the interaction the event created. It's important to respond to an interaction within 3 seconds, or it will time out. Responding after this period will throw a `NotFoundException`.
```

Comment:
> In the event args, `Id` will be the id of the button you specified.

The ID is the ID? Thank u velveeta.

Reply:
` Id` to discord actually refers to the snowflake of the interaction tyvm